frontend/src/component/feature/FeatureStrategy/FeatureStrategyEdit/FeatureStrategyEdit.tsx

```diff
@@ -90,7 +90,7 @@ export const FeatureStrategyEdit = () => {
     useEffect(() => {
         // Fill in the selected segments once they've been fetched.
         savedStrategySegments && setSegments(savedStrategySegments);
-    }, [savedStrategySegments]);
+    }, [JSON.stringify(savedStrategySegments)]);
```

Comment:
when we fixed useSegments it was returning [] multiple times (confirmed in console.log). but it was always a different instance of empty array